common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -808,6 +808,34 @@ def index_dictionary(self):
 
         return xblock_body
 
+    def get_children_discussion_toggle_status(self):
+        """
+        Yield values of children's discussion_enabled flag
+        """
+        for child in self.get_children():
+            if hasattr(child, 'discussion_enabled'):
+                yield child.discussion_enabled
+            elif hasattr(child, 'get_children_discussion_toggle_status'):
+                yield from child.get_children_discussion_toggle_status()
+
+    def get_discussion_toggle_status(self):
```

Comment:
We can consider removing this entirely and just showing authors in the UI which units have discussions enabled via an icon. We can still have a helper that bulk-enables discussions under a section/subsection, without pre-computing the status.

Reply:
Okay, unless we have a compelling use case to utilize it, lets drop it.
The recursive _set_ makes sense in these cases, but the recursive _get_ seems out of place.

Performance aside, I'm uneasy about calling a course toggle "disabled" just because it's not actively being used in any of the children. While the UI may be equivalent at the unit level (whether the feature is off for the course or off for the unit, the display is the same), it could become confusing elsewhere, eg:
- create a course
- enable toggle (before adding units)
- notice that the UI now says "feature is disabled for the course"; this would be confusing if I had just enabled it...
  - assuming we use the flag for some kind of display like that ^; and if we're _not_ ever displaying it anywhere like that, we don't need it ;)